native/src/modules/layout/containers/HeaderContainer.js

```diff
@@ -97,17 +95,16 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
   }
 
   const peeking = isPeekingRoute(state, { routeCity: route.city })
-  const language = route.language
-  const path = route.path || undefined
-  const newsPath = [route.type || null, route.newsId || null].filter(Boolean).join('/')
-  const shareUrl = path
-    ? url(path)
-    : cityContentUrl({
-        cityCode: route.city,
-        languageCode: route.language,
-        route: routeName,
-        path: newsPath
-      })
+  const { routeType, language, city } = route
+
+  const routeInformation: RouteInformationType =
+    routeType === NEWS_ROUTE
+      // $FlowFixMe Flow does not get anything and hates the whole world :/ I am already checking the routeType!
+      ? { route: NEWS_ROUTE, languageCode: language, cityCode: city, newsType: route.type, newsId: route.newsId }
```

Comment:
to fix the flow error check the RouteInformationType. `newsId` does not seem to be nullable, but is for the route mapping type.

Reply:
also to help flow write `route.routeType === NEWS_ROUTE`

Reply:
Another good suggestion :D thanks :+1: The last fixme is only solvable with a lot more complicated code which is not worth it imo as cityContentPath has to be defined for category/dashboard routes but is optional for events/pois.

Reply:
Or do you have any more tips for me @Taggotty :P